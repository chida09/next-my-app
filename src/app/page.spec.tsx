import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import Home from './page';
import axios from 'axios';
import userEvent from '@testing-library/user-event';

jest.mock('axios');
const mockedAxios = axios as jest.Mocked<typeof axios>;

describe('Home component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render', () => {
    const { asFragment } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
  })

  it('should display "登録しました" when registration is successful', async () => {
    mockedAxios.post.mockResolvedValueOnce({
      data: {},
      status: 200,
      statusText: 'OK',
      headers: {},
      config: {},
    });
    render(<Home />);
    const titleInput = screen.getByPlaceholderText('Enter title');
    const saveButton = screen.getByText('Save');

    await userEvent.type(titleInput, 'Test title');
    await userEvent.click(saveButton);

    await waitFor(() => {
      expect(screen.getByText('登録しました')).toBeInTheDocument();
      expect(screen.queryByText('登録に失敗しました')).not.toBeInTheDocument();
    });

    expect(mockedAxios.post).toHaveBeenCalledTimes(1);
    expect(mockedAxios.post).toHaveBeenCalledWith('https://jsonplaceholder.typicode.com/posts', {
      title: 'Test title',
      body: 'bar',
      userId: 1,
    });
  });

  it('should display "登録に失敗しました" when registration fails', async () => {
    mockedAxios.post.mockResolvedValueOnce({
      data: {},
      status: 500,
      statusText: 'Internal Server Error',
      headers: {},
      config: {},
    });

    render(<Home />);
    const titleInput = screen.getByPlaceholderText('Enter title');
    const saveButton = screen.getByText('Save');

    await userEvent.type(titleInput, 'Test title');
    await userEvent.click(saveButton);

    await waitFor(() => {
      expect(screen.getByText('登録に失敗しました')).toBeInTheDocument();
      expect(screen.queryByText('登録しました')).not.toBeInTheDocument();
    });

    expect(mockedAxios.post).toHaveBeenCalledTimes(1);
    expect(mockedAxios.post).toHaveBeenCalledWith('https://jsonplaceholder.typicode.com/posts', {
      title: 'Test title',
      body: 'bar',
      userId: 1,
    });
  });
});
