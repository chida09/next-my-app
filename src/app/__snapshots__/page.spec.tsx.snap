// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should render 1`] = `
<DocumentFragment>
  <main>
    <form>
      <input
        placeholder="Enter title"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Save
      </button>
    </form>
  </main>
</DocumentFragment>
`;
